### PR TITLE
Issue #11: [Vendor payouts] Refactor code to get pending payouts as per clients and currencies

### PR DESCRIPTION
**Pull request by deepa-g**
…cular vendor then that vendor should not be deleted

### CODE REVIEW

File: src/com_tjvendors/admin/models/vendors.php [hunk 1]
```diff
@@ -208,6 +208,32 @@ public function deleteClientFromVendor($vendor_id,$client)
 		$tjvendorsModelVendor = JModelLegacy::getInstance('Vendor', 'TjvendorsModel');
 		$vendorData           = $tjvendorsModelVendor->getItem($vendor_id);
 		$db                   = $this->getDbo();
+		
+		//Get vendor specific Fees Data
+		$query = $db->getQuery(true);
```
Comment (coolbung) [inline src/com_tjvendors/admin/models/vendors.php]: Need to use the `getItems()` from the fees model.

File: src/com_tjvendors/admin/models/vendors.php [hunk 2]
```diff
@@ -208,6 +208,32 @@ public function deleteClientFromVendor($vendor_id,$client)
...
+		$query->where($db->quoteName('client') . ' = ' . $db->quote($client));
+		$db->setQuery($query);
+		$VendorFeeData = $db->loadAssocList();
+		
+		if (!empty($VendorFeeData))
+		{
+			foreach ($VendorFeeData as $FeeData)
+			{
+				// Getting Vendor Payable Amount here
+				$result = TjvendorsHelper::getPayableAmount($FeeData['vendor_id'], $FeeData['client'], $FeeData['currency']);
```
Comment (coolbung) [inline src/com_tjvendors/admin/models/vendors.php]: What is this method doing ?

There should be a single method in the vendor model that gets the outstanding amount for a single vendor. That is the method you should be simply calling once, not even in a loop.

Comment (deepa-g) [reply src/com_tjvendors/admin/models/vendors.php]: @coolbung 
Do you mean the function will return the array of 'currency' => payable amount?

File: src/com_tjvendors/admin_language/en-GB/en-GB.com_tjvendors.ini [hunk 3]
```diff
@@ -236,3 +236,6 @@ COM_TJVENDORS_SAVE_ALIAS_WARNING="Alias already existed so a number was added at
 COM_TJVENDOR_GENERIC_CLIENT="All"
 ;Added in 1.3.0
 COM_TJVENDORS_FORM_LBL_VENDOR_PAYMENT_GATEWAY_DESC="Choose the payment gateway which you want to configure."
+
+;Added in
```
Comment (manojLondhe) [inline src/com_tjvendors/admin_language/en-GB/en-GB.com_tjvendors.ini]: added in ?

File: src/com_tjvendors/admin/models/vendors.php [hunk 4]
```diff
@@ -205,10 +205,33 @@ public function deleteVendor($vendor_id)
...
+		// Get vendor specific Fees Data
+		$tjvendorsModelVendorFees->setState('vendor_id', $vendor_id);
+		$vendorFeeData            = $tjvendorsModelVendorFees->getItems();
+
+		if (!empty($vendorFeeData))
+		{
+			foreach ($vendorFeeData as $feeData)
+			{
+				// Getting Vendor Payable Amount here
+				$result = TjvendorsHelper::getPayableAmount($feeData->vendor_id, $feeData->client, $feeData->currency);
```
Comment (manojLondhe) [inline src/com_tjvendors/admin/models/vendors.php]: @coolbung This is the method to get the balance payable to a vendor for given client and currency. Are you saying we need method that outputs remaining payouts for all clients and all currencies?

Comment (coolbung) [reply src/com_tjvendors/admin/models/vendors.php]: Yes. My suggestion is

1. Create a `getPayableAmount()` method in the vendor (singular) model with client and currency as optional inputs. Currently its in a helper which is not correct. 
2. The method always returns an array of objects and each object has the client, amount and currency.
3. If client and currency are optionally passed the array that is returned is appropriately filtered. 

This should be the one and only way to get the balance for a single vendor.

Comment (manojLondhe) [reply src/com_tjvendors/admin/models/vendors.php]: sounds good

File: src/com_tjvendors/admin_language/en-GB/en-GB.com_tjvendors.ini (hunk 3 above)
Comment (deepa-g) [reply src/com_tjvendors/admin_language/en-GB/en-GB.com_tjvendors.ini]: Resolved

File: src/com_tjvendors/admin/models/vendors.php (hunk 1 above)
Comment (deepa-g) [reply src/com_tjvendors/admin/models/vendors.php]: Resolved

File: src/com_tjvendors/admin/models/vendors.php (hunk 2 above)
Comment (deepa-g) [reply src/com_tjvendors/admin/models/vendors.php]: Resolved

File: src/com_tjvendors/site/models/vendor.php [hunk 5]
```diff
@@ -525,4 +525,110 @@ public function formatPaymentStructure($data, $paymentDetails)
...
 	}
+	
+	/**
+	 * Get get vendor_id
+	 *
+	 * @param   integer  $vendor_id  integer
+	 * @param   string   $client     string like com_jgive
+	 * @param   string   $currency   string like USD, EUR
+	 *
+	 * @return integer
```
Comment (manojLondhe) [inline src/com_tjvendors/site/models/vendor.php]: this is array

File: src/com_tjvendors/site/models/vendor.php [hunk 6]
```diff
@@ -525,4 +525,110 @@ public function formatPaymentStructure($data, $paymentDetails)
...
+			$query->where($db->quoteName('client') . ' = ' . $db->quote($client));
+		}
+		
+		$query->where($db->quoteName('status') . ' = ' . $db->quote(1));
+		$query->group($db->quoteName('currency'));
+		$db->setQuery($query);
+		$debit = $db->loadAssocList();
+
+		// Total credit amount against the vendor for e.g. 100 ($50 + €50)
+		if (!empty($credit))
```
Comment (manojLondhe) [inline src/com_tjvendors/site/models/vendor.php]: handle else

File: src/com_tjvendors/site/models/vendor.php (hunk 5 above)
Comment (deepa-g) [reply src/com_tjvendors/site/models/vendor.php]: Updated

File: src/com_tjvendors/site/models/vendor.php (hunk 6 above)
Comment (deepa-g) [reply src/com_tjvendors/site/models/vendor.php]: Updated

**Pull request merged**